horreum-backend/src/main/java/io/hyperfoil/tools/horreum/entity/user/UserApiKey.java

```diff
@@ -0,0 +1,164 @@
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.NamedQueries;
+import jakarta.persistence.NamedQuery;
+import jakarta.persistence.SequenceGenerator;
+import jakarta.persistence.Table;
+import jakarta.persistence.Transient;
+
+import io.hyperfoil.tools.horreum.api.internal.services.UserService;
+import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
+
+@Entity
+@Table(name = "userinfo_apikey")
+@NamedQueries({
+        // fetch all keys that expire on a given day
+        @NamedQuery(name = "UserApiKey.expire", query = "from UserApiKey where not revoked AND (access is null and (creation + active day) = ?1 or (access + active day) = ?1)"),
+        // fetch all keys that have gone past their expiration date
+        @NamedQuery(name = "UserApiKey.pastExpiration", query = "from UserApiKey where not revoked AND (access is null and (creation + active day) < ?1 or (access + active day) < ?1)"),
+})
+public class UserApiKey extends PanacheEntityBase implements Comparable<UserApiKey> {
+
+    // old authentication tokens are not listed and can't be modified, but are kept around to prevent re-use
+    public static long ARCHIVE_AFTER_DAYS = 7;
+
+    @Id
+    @SequenceGenerator(name = "apikeyIdGenerator", sequenceName = "userinfo_apikey_id_seq", allocationSize = 1)
+    @GeneratedValue(strategy = SEQUENCE, generator = "apikeyIdGenerator")
+    public long id;
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "username")
+    public UserInfo user;
+
+    @Transient
+    private final UUID randomnessSource;
+
+    private final String hash;
+
+    public String name;
+
+    @Enumerated
+    public final UserService.KeyType type;
+
+    public LocalDate creation, access;
+
+    public long active; // number of days after last access that the key remains active
+
+    public boolean revoked;
+
+    public UserApiKey() {
+        randomnessSource = null;
+        hash = null;
+        name = null;
+        type = UserService.KeyType.USER;
+    }
+
+    public UserApiKey(UserService.ApiKeyRequest request, LocalDate creation, long valid) {
+        this(request.name, request.type, creation, valid);
+    }
+
+    public UserApiKey(String name, UserService.KeyType type, LocalDate creationDate, long valid) {
+        randomnessSource = UUID.randomUUID();
+        this.name = name == null ? "" : name;
+        this.type = type;
+        this.active = valid;
+        hash = computeHash(keyString());
+        creation = creationDate;
+        revoked = false;
+    }
+
+    public UserService.ApiKeyResponse toResponse() {
+        UserService.ApiKeyResponse response = new UserService.ApiKeyResponse();
+        response.id = id;
+        response.name = name;
+        response.type = type;
+        response.creation = creation;
+        response.access = access;
+        response.isRevoked = revoked;
+        response.toExpiration = toExpiration(LocalDate.now());
+        return response;
+    }
```

Comment:
For all other entities, mapping has been decoupled from the API via Mappers defined in the `io.hyperfoil.tools.horreum.mapper` package